### ISSUE TITLE
onnx model?

### ISSUE THREAD

**Issue by PhenomenalOnee** [lucidrains/x-transformers#28]
onnx conversion error:
triu operator not supported in line (for attention):
if.causal:
mask = torch.zeros((i,j)).triu_(j - i + 1).bool()

Any Help

**Comment by lucidrains** [comment]
@PhenomenalOnee I replaced `triu_` in https://github.com/lucidrains/x-transformers/commit/6119be8d815b9ee2eb01c08fdaa49190b83ab03d and it should be onnx compatible now!

**Comment by minhnvl** [comment]
I'm getting the error for converting to ONNX model. The error appears when I use onnxruntime load model with the content: "Type Error: Type 'tensor(bool)' of input parameter (onnx::Mul_1667) of operator (Mul) in node (Mul_1289) is invalid." 

After tracking, I saw the error related to TORCH.TENSOR.MASKED_FILL doesn't support from onnxruntime. The related code in line 435, x_transformer.py version 0.15.0. 

Could you help me solve it?

**Comment by bitdom8** [comment]
@PhenomenalOnee hi, can you help me solving this? What have you used as code? this one?

```
    torch.onnx.export(model,               # model being run
            torch.from_numpy(train_x),                         # model input (or a tuple for multiple inputs)
            "super_resolution.onnx",)
```

**Comment by sinpy1117** [comment]
> I'm getting the error for converting to ONNX model. The error appears when I use onnxruntime load model with the content: "Type Error: Type 'tensor(bool)' of input parameter (onnx::Mul_1667) of operator (Mul) in node (Mul_1289) is invalid."
> 
> ![image](https://user-images.githubusercontent.com/56904021/184310103-31a596ad-f9a6-4a4f-8ad9-5a7de9c1dfbe.png)
> 
> After tracking, I saw the error related to TORCH.TENSOR.MASKED_FILL doesn't support from onnxruntime. The related code in line 435, x_transformer.py version 0.15.0.
> 
> Could you help me solve it?

hi,have you solved this? I met the same problem ~
